src/transformStateWithClones.js

```diff
@@ -7,7 +7,44 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  // Here I clone the given state / создаю объект текущего состояния
+  let currentState = { ...state };
+
+  for (const action of actions) {
+    const type = action.type;
+    let nextState;
+
+    /* Here I checked action condition for every case
+    and implement what I need to do
+    проверяю состояние действия для каждого случая
+    и выполняю действия в зависимости от условия */
+
+    if (type === 'addProperties') {
```

Comment:
Describe all variants of cases in constant.
const ACTION_CASES = {
add: 'addProperties',
}